In [13]:
def new_function(num):
    new_list= []
    x=[]
    if num == 0:
        num = 111111
    if num > 0:
        num= str(num)
    for number in num:
        new_list.append(number)
        new_list.sort(reverse=True)
        x= ''.join([num for num in new_list])
    print(x)

In [15]:
new_function(68436419)
new_function(0)

98664431
111111


2.

In [2]:
import pandas as pd
jeopardy_path = "JEOPARDY_CSV.csv"
jeopardy_df = pd.read_csv(jeopardy_path)
jeopardy_df

Show Number    Air Date             Round  \
0              4680  12/31/2004         Jeopardy!   
1              4680  12/31/2004         Jeopardy!   
2              4680  12/31/2004         Jeopardy!   
3              4680  12/31/2004         Jeopardy!   
4              4680  12/31/2004         Jeopardy!   
...             ...         ...               ...   
216925         4999   5/11/2006  Double Jeopardy!   
216926         4999   5/11/2006  Double Jeopardy!   
216927         4999   5/11/2006  Double Jeopardy!   
216928         4999   5/11/2006  Double Jeopardy!   
216929         4999   5/11/2006   Final Jeopardy!   

                               Category    Value  \
0                               HISTORY    $200    
1       ESPN's TOP 10 ALL-TIME ATHLETES    $200    
2           EVERYBODY TALKS ABOUT IT...    $200    
3                      THE COMPANY LINE    $200    
4                   EPITAPHS & TRIBUTES    $200    
...                                 ...      ...   
216925                   RIDDLE ME THIS  $2,000    
216926                        "T" BIRDS  $2,000    
216927           AUTHORS IN THEIR YOUTH  $2,000    
216928                       QUOTATIONS  $2,000    
216929                   HISTORIC NAMES     None   

                                                 Question  \
0       For the last 8 years of his life, Galileo was ...   
1       No. 2: 1912 Olympian; football star at Carlisl...   
2       The city of Yuma in this state has a record av...   
3       In 1963, live on "The Art Linkletter Show", th...   
4       Signer of the Dec. of Indep., framer of the Co...   
...                                                   ...   
216925  This Puccini opera turns on the solution to 3 ...   
216926  In North America this term is properly applied...   
216927  In Penny Lane, where this "Hellraiser" grew up...   
216928  From Ft. Sill, Okla. he made the plea, Arizona...   
216929  A silent movie title includes the last name of...   

                                Answer  
0                           Copernicus  
1                           Jim Thorpe  
2                              Arizona  
3                           McDonald's  
4                           John Adams  
...                                ...  
216925                        Turandot  
216926                      a titmouse  
216927                    Clive Barker  
216928                        Geronimo  
216929  Grigori Alexandrovich Potemkin  

[216930 rows x 7 columns]

a. There are 216930 rows, they are the total number of questions asked between 12/31/2004 and 05/11/2006

In [20]:
unique_shows= jeopardy_df.groupby('Show Number').count()
print(unique_shows)

              Air Date   Round   Category   Value   Question   Answer
Show Number                                                          
1                   48      48         48      48         48       48
2                   50      50         50      50         50       50
3                   51      51         51      51         51       51
4                   53      53         53      53         53       53
5                   54      54         54      54         54       54
...                ...     ...        ...     ...        ...      ...
6295                58      58         58      58         58       58
6296                61      61         61      61         61       61
6297                59      59         59      59         59       59
6298                61      61         61      61         61       61
6300                30      30         30      30         30       30

[3640 rows x 6 columns]


b. 3640 unique shows 
c. If there were any repeat questions in those two years